contracts/PanopticPool.sol

```diff
@@ -1242,95 +1246,90 @@ contract PanopticPool is ERC1155Holder, Multicall {
 
     /// @notice Liquidates a distressed account. Will burn all positions and will issue a bonus to the liquidator.
     /// @dev Will revert if: account is not margin called or if the user liquidates themselves.
-    /// @param account Address of the distressed account.
-    /// @param tickLimitLow The lower tick slippagelimit.
-    /// @param tickLimitHigh The upper tick slippagelimit.
+    /// @param liquidatee Address of the distressed account.
     /// @param positionIdList List of positions owned by the user. Written as [tokenId1, tokenId2, ...].
-    function liquidateAccount(
-        address account,
-        int24 tickLimitLow,
-        int24 tickLimitHigh,
-        uint256[] calldata positionIdList
+    function liquidate(
+        address liquidatee,
+        uint256[] calldata positionIdList,
+        uint256 delegation0,
+        uint256 delegation1
     ) external {
-        // validate positionIdList (list of all tokenIds of the account)
-        _validatePositionList(account, positionIdList, 0);
+        _validatePositionList(liquidatee, positionIdList, 0);
 
-        // check slippage limits, read current price + tick
-        int24 currentTick;
-        (currentTick, , tickLimitLow, tickLimitHigh) = _getPriceAndCheckSlippageViolation(
-            tickLimitLow,
-            tickLimitHigh
-        );
+        if (numberOfPositions(msg.sender) > 0) revert Errors.LiquidatorHasOpenPositions();
 
-        // bonusAmounts the amount of liquidation bonuses going to the liquidator
-        // delegatedAmounts delegate amount that is required to cover the collateral requirements for the entirety of the portfolio (do not consider the existing balance)
-        int256 bonusAmounts;
-        uint256 delegatedAmounts;
+        // Assert the account we are liquidating is actually insolvent
+        int24 twapTick = getUniV3TWAP();
 
+        // While the liquidation bonus is based on the amount of tokens the liquidator was forced to convert,
+        // they also receive a basal bonus (1% of col. req. for liquidated positions)
+        // in exchange for completing a valid liquidation, even if they did not have to convert any tokens
+        // and their variable bonus is 0
+        uint256 basalBonus0; // 1% of collateral requirement
+        uint256 basalBonus1; // 1% of collateral requirement
         {
-            // Compute all premia (long and short)
+            (, int24 currentTick, , , , , ) = s_univ3pool.slot0();
+
+            // Enforce maximum delta between TWAP and currentTick to prevent extreme price manipulation
+            if (Math.abs(int256(currentTick) - int256(twapTick)) > MAX_TWAP_DELTA_LIQUIDATION)
+                revert Errors.StaleTWAP();
+
             (int256 premia, uint256[2][] memory positionBalanceArray) = _calculateAccumulatedPremia(
-                account,
+                msg.sender,
```

Comment:
Why `msg.sender` here and not `liquidatee`?

Reply:
Good catch. Actually, I'm thinking of not including the premia here at all. The 1% bonus is rather arbitrary, and since premia is usually a small component of the collateral requirement anyway, there's no need to make things more complex by calculating it here.

Reply:
We do need it for the solvency check though, so may need to leave it in